example_generator/generator.py

```diff
@@ -0,0 +1,200 @@
+        self.__patch_parse(JSF)
+
+    def random_example(self, schema):
+        """Create random example for given schema"""
+        fake_data_object = JSF(schema)
+        fake_json = fake_data_object.generate()
+        self._handle_if_then(fake_json, schema)
+        self._handle_dependant_schemas(fake_json, schema)
+        return fake_json
+
+    def _handle_dependant_schemas(self, fake_json, json_schema):
+        """jsf library doesn't handle dependent schemas yet"""
+        if 'dependentSchemas' in json_schema:
+            dependencies = (set(fake_json.keys()).intersection(
+                set(json_schema['dependentSchemas'])))
+            for dep_schema_key in dependencies:
+                dep_schema = (
+                    json_schema['dependentSchemas'][dep_schema_key])
+                dep_schema['type'] = 'object'
+                dep_fake_json = self.random_example(dep_schema)
+                fake_json.update(dep_fake_json)
+
+    def _handle_if_then(self, fake_json, json_schema):
+        """jsf library doesn't handle if then properties yet"""
+        if 'allOf' in json_schema:
+            for all_of_item in json_schema['allOf']:
+                if 'if' in all_of_item:
+                    item_key = list(all_of_item['if']['properties'].keys())[0]
+                    item_value = (
+                        all_of_item['if']['properties'][item_key]['const'])
+                    if(item_key in fake_json and
+                            fake_json[item_key] == item_value):
+                        dep_schema = all_of_item['then']
+                        dep_schema['type'] = 'object'
+                        if '$defs' in json_schema:
+                            dep_schema['$defs'] = json_schema['$defs']
+                        dep_json = self.random_example(dep_schema)
+                        self._handle_dep_refs(all_of_item,
+                                              json_schema,
+                                              dep_json)
+                        fake_json.update(dep_json)
+        return fake_json
+
+    def _handle_dep_refs(self, all_of_item, json_schema, dep_json):
+        """jsf library doesn't handle if then in array items"""
+        for prop_key in all_of_item['then']['properties']:
+            prop_value = all_of_item['then']['properties'][prop_key]
+            if 'items' in prop_value and '$ref' in prop_value['items']:
+                ref_key_str = prop_value['items']['$ref']
+                ref_key = ref_key_str.replace('#/', '').split('/')
+                dict_item = reduce(operator.getitem, ref_key, json_schema)
+                if 'allOf' in dict_item:
+                    new_array = []
+                    for json_item in dep_json[prop_key]:
+                        self._handle_if_then(json_item, dict_item)
+                        new_array.append(json_item)
+                    dep_json[prop_key] = new_array
+
+    @staticmethod
+    def __patch_generate(cls):
```

Comment:
A function with a leading double-underscore gets name mangled to prevent accidental access. For example on line 119, in order to patch the `__parse_definitions` method from the jsf library, I had to access it as `_JSF__parse_definitions`. My idea is to use these monkey patches as a temp fix and maybe work with the jsf maintainers to fix things there.